src/data_classifier.py

```diff
@@ -0,0 +1,173 @@
+    stdstar = {}
+    science = {}
+
+    for ob in obs:
+        fn = data_dir + ob + '.fits'
+        f = pyfits.open(fn)
+        imagetype = f[0].header['IMAGETYP'].upper()
+        obj_name = f[0].header['OBJECT']
+        f.close()
+        #---------------------------
+        # check if it is within a close distance to a standard star
+        # if so, fix the object name to be the good one from the list!
+        try:
+            near_std, std_dist = wifes_calib.find_nearest_stdstar(fn)
+            if std_dist < 100.0:
+                obj_name = near_std
+        except:
+            pass
+        #---------------------------
+        # 1 - bias frames
+        if imagetype == 'BIAS':
+            bias.append(ob)
+        # 2 - quartz flats
+        if imagetype == 'FLAT':
+            domeflat.append(ob)
+        # 3 - twilight flats
+        if imagetype == 'SKYFLAT':
+            twiflat.append(ob)
+        # 4 - dark frames
+        if imagetype == 'DARK':
+            dark.append(ob)
+        # 5 - arc frames
+        if imagetype == 'ARC':
+            arc.append(ob)
+        # 6 - wire frames
+        if imagetype == 'WIRE':
+            wire.append(ob)
+        # 7 - standard star
+        if imagetype == 'STANDARD':
+            # group standard obs together!
+            if obj_name in stdstar.keys():
+                stdstar[obj_name].append(ob)
+            else:
+                stdstar[obj_name] = [ob]
+        
+        # all else are science targets (also consider standar star in imagety = OBJECT)
+        if imagetype == 'OBJECT':
+            if obj_name in stdstar_list:
+                # group standard obs together!
+                if obj_name in stdstar.keys():
+                    stdstar[obj_name].append(ob)
+                else:
+                    stdstar[obj_name] = [ob]
+            else:
+                # group science obs together!
+                if obj_name in science.keys():
+                    science[obj_name].append(ob)
+                else:
+                    science[obj_name] = [ob]
+
+    # #------------------
+    # science dictionay
+
+    sci_obs = []
+
+    for obj_name in science.keys():
+        obs_list = science[obj_name]
+        sci_obs.append({'sci':obs_list, 'sky':[]})
+
+
+    #------------------
+    # stdstars dictionary
+    std_obs = []
+
+    for obj_name in stdstar.keys():
+        obs_list = stdstar[obj_name]
+        std_obs.append({'sci':obs_list, 'name':obj_name,'type':['flux', 'telluric']})
+
+
+    obs_metadata = {
+        'bias' : bias,
+        'domeflat' : domeflat,
+        'twiflat' : twiflat,
+        'dark' : dark,
+        'wire' : wire,
+        'arc'  : arc,
+        'sci'  : sci_obs,
+        'std'  : std_obs}
+    
+    return obs_metadata 
+
+
+
+
+
+
+def classify(data_dir):
+
+    # What is this option for? TODO change to a default variable in the function
+    try:
+        naxis2_use = int(sys.argv[2])
+    except:
+        naxis2_use = 0
```

Comment:
Good, thank you! I think I understand what you meant but please, take a look at the changes I did in both reduce_data.py and data_classifier.py. 
Apparently, the `naxis2_use = int(sys.argv[2])` was an option added to enable stellar mode or non-stellar mode via NAXIS2, so I'm not totally sure if we'll need to have that option in the automated pipeline. I'll leave it just in case and add it to my list of checks to do/ask about regarding specific observing modes/configurations.

Reply:
You've done exactly what I asked, I'm just not sure about the variable names I asked for! 😅 

Does `naxis2_to_process` make sense? I was assuming it could be 0, or 1, or 2, etc. depending on some `naxis2` field.

If it's more of a boolean flag, I think `use_naxis2` would be better?

Reply:
Oh sorry, I asked you to make a bug too, `naxis2_to_process` needs to default to 0.

So in `reduce_data.py`:
```python

naxis2_to_process = 0
if len(sys.argv) == 3:
    naxis2_to_process = int(sys.argv[2])
```

Reply:
Fixed now. I thought I was missing something as `naxis2_to_process` does not always have a value in the previous version but wasn't sure if I was understanding the whole process. Now I do, although still not sure about how useful is this. I'll check it out for follow-ups.